### PR TITLE
Global Styles on Personal: Add feature check and sticker to eligible blogs

### PR DESCRIPTION
**Pull request by rcrdortiz**
Fixes:
- https://github.com/Automattic/dotcom-forge/issues/9518

## Proposed changes:

* Added a new check before granting access to GS
* This new check makes sure that the blog is eligible to access GS under the new circumstances (blog older than xxx and on a Personal plan)
* If access is granted, the `wpcom-global-styles-personal-plan` is added to the site.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* In your `0-sandbox.php` file add: `const GLOBAL_STYLES_ON_PERSONAL_PLAN = true;`
* Now all new sites you create with the Personal plan should have access to the feature.
* Make sure there are no regressions on Atomic sites.
* Make sure that on old sites the Premium plan is required.
* The different upgrade strategies are excluded from this PR, the upgrade wording will still reference the Premium plan, but you should stop seeing them if you have a Personal plan.
* Remove the feature flag (set it to false) and you should still be able to use GS
* Remove the blog sticker and you should be prompted to upgrade to Premium

https://github.com/user-attachments/assets/a0017cad-35d0-4ea0-ad8b-2cc11deda57f

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php
Comment (BogdanUngureanu) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php:640]: I think this needs to be generic so we can reuse it in the logic for custom CSS addon; I think the sticker should be re-used there.

Comment (rcrdortiz) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php:640]: Let's not fall into early generalization. Especially considering there is a 99.99% chance of not reverting. We have an issue for handling the Custom CSS add-on ([here](https://github.com/Automattic/dotcom-forge/issues/9519))

Comment (BogdanUngureanu) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php:640]: Yeah, I'm working on that issue and realized I needed something similar for the custom CSS too. :) 

I'll copy your code so that we don't block this one and do a refactor after these two implementations land. Sounds good? :)

Comment (rcrdortiz) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-global-styles/index.php:640]: I doubt we need a refactor, bear in mind we're going to delete this code eventually.

**Review by dsas — APPROVED** [review]
Works nicely for me on simple. Doesn't have any effect on atomic, as expected 👍

**Pull request merged**
